### ISSUE TITLE
aaa

### ISSUE THREAD

**Issue by choisekyu** [choisekyu/aaa#1]
import tensorflow as tf
import numpy as np
from tensorflow.examples.tutorials.mnist import input_data
import random

import matplotlib # to plot images
# Force matplotlib to not use any X-server backend.
matplotlib.use('Agg')
import matplotlib.pyplot as plt
import matplotlib.gridspec as gridspec

n_visible = 6
n_hidden = 3
corruption_level = 0.3

# create node for input data
X = tf.placeholder("float", [None, n_visible], name='X')

# create node for corruption mask
mask = tf.placeholder("float", [None, n_visible], name='mask')

# create nodes for hidden variables
W_init_max = 4 * np.sqrt(6. / (n_visible + n_hidden))
W_init = tf.random_uniform(shape=[n_visible, n_hidden],
                           minval=-W_init_max,
                           maxval=W_init_max)

W = tf.Variable(W_init, name='W')
b = tf.Variable(tf.zeros([n_hidden]), name='b')

W_prime = tf.transpose(W)  # tied weights between encoder and decoder
b_prime = tf.Variable(tf.zeros([n_visible]), name='b_prime')

def model(X, mask, W, b, W_prime, b_prime):
    tilde_X = mask * X  # corrupted X

    Y = tf.nn.sigmoid(tf.matmul(tilde_X, W) + b)  # hidden state
    Z = tf.nn.sigmoid(tf.matmul(Y, W_prime) + b_prime)  # reconstructed input
    return Z

# build model graph
Z = model(X, mask, W, b, W_prime, b_prime)

# create cost function
cost = 0.5 * tf.reduce_sum(tf.pow(X - Z, 2))  # minimize squared error
train_op = tf.train.GradientDescentOptimizer(0.05).minimize(cost)  # construct an optimizer
predict_op = Z

input_arm = np.loadtxt("data/force_dataset_in.csv",delimiter=",")
arm_max = input_arm.max()
trX = input_arm[10000:] / arm_max
teX = input_arm[:10000] / arm_max
#trX = input_arm
#teX=input_arm

# Launch the graph in a session
with tf.Session() as sess:
    # you need to initialize all variables
    tf.initialize_all_variables().run()

    for i in range(5):
        for start, end in zip(range(0, len(trX), 100), range(100, len(trX)+1, 100)):
            input_ = trX[start:end]
            mask_np = np.random.binomial(1, 1 - corruption_level, input_.shape)
            sess.run(train_op, feed_dict={X: input_, mask: mask_np})

        mask_np = np.random.binomial(1, 1 - corruption_level, teX.shape)
        print(i, sess.run(cost, feed_dict={X: teX, mask: mask_np}))

    # save the predictions for 100 images
    num = 5
    mask_np = np.random.binomial(1, 1 - corruption_level, teX[:num].shape)
    predicted_imgs = sess.run(predict_op * teX[:num].max(), feed_dict={X: teX[:num], mask: mask_np}) * arm_max
    input_imgs = teX[:num] * arm_max

    print input_imgs, '\n'
    print predicted_imgs
    print('Done')
